sys/include/stdio_nimble.h

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (C) 2019 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    sys_stdio_nimble STDIO over UART + NimBLE
+ * @ingroup     sys
+ *
+ * @brief       Standard input/output backend using UART and NimBLE.
+ *              Define STDIO_NIMBLE_DISABLE_UART in CFLAGS to disable the UART part.
+ *
+ * @{
+ * @file
+ *
+ * @author      Hendrik van Essen <hendrik.ve@fu-berlin.de>
+ */
+
+#ifndef STDIO_NIMBLE_H
+#define STDIO_NIMBLE_H
+
+#include "stdio_base.h"
+
+#include "nimble_riot.h"
+#include "net/bluetil/ad.h"
+
+#include "host/ble_hs.h"
+#include "host/util/util.h"
+#include "host/ble_gatt.h"
+#include "services/gap/ble_svc_gap.h"
+#include "services/gatt/ble_svc_gatt.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Nimble handle for stdout characteristic
+ */
+static uint16_t _val_handle_stdout;
+
+/**
+ * @brief UUID for stdio service (value: e6d54866-0292-4779-b8f8-c52bbec91e71)
+ */
+static const ble_uuid128_t gatt_svr_svc_stdio_uuid
```

Comment:
should better be allocated in the C file. If they should be available to other modules, I a define in the header is the better choice.

Having it like this would mean the compiler could possibly allocated this memory once for every compilation unit this header is included...

Reply:
This header wasn't meant to be included by another module, as stdio_nimble is initiated by the auto_init mechanism. I moved these parts out anyway.